include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_dual_splitting.cpp

```diff
@@ -740,11 +740,11 @@ TimeIntBDFDualSplitting<dim, Number>::viscous_step()
 
     // solve linear system of equations
     bool const update_preconditioner =
-      this->param.update_preconditioner_viscous and
-      ((this->time_step_number - 1) % this->param.update_preconditioner_viscous_every_time_steps ==
+      this->param.update_preconditioner_momentum and
```

Comment:
This current form only realizes the linear version of the scheme, this is not what you described in the first comment, is it? Can you point me to the code section where it does the implicit convective term minus extrapolated convetive term (I would expect a modification of the rhs, adding the extrapolation)? The linear momentum operator contains an extrapolated convective term, right? Is that why you do not consider for the extrapolation on the rhs?
In a follow-up PR we should allow for a Newton solver here, then it would actually need the subtraction, right? ... also we should update the potentially nonlinear viscous  terms as we did in the pressure-correction scheme while we are at it ;)

@nfehn please explain a bit more :)

Reply:
The convective term is nonlinear and a Newton scheme is used. In the function `rhs_viscous()`, the extrapolated convective term shows up.